src/System.Private.CoreLib/shared/System/Threading/SemaphoreSlim.cs

```diff
@@ -48,17 +49,17 @@ public class SemaphoreSlim : IDisposable
         private int m_countOfWaitersPulsedToWake;
 
         // Dummy object used to in lock statements to protect the semaphore count, wait handle and cancelation
-        private object m_lockObj;
+        private object? m_lockObj; // initialized non-null, then set to null on Dispose
```

Comment:
I think it's unfortunate that we have this design of nulling out the lock object; ideally this wouldn't only be non-nullable, it'd be readonly.  But I'm not going to change any of that now as part of this nullability work.

Reply:
As far as I can tell, it could be done in a safe and unobservable way. All public entry points check for dispose. Maybe worth a `// TODO-NULLABLE` for later (since it would remove the `?`)